integration-tests/itest/deploy_test.go

```diff
@@ -0,0 +1,461 @@
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should produce a counter in the flink output log", func() {
+			err := checkCountInLog(swissKnifeApp, "flink-egress")
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+})
+
+func checkCountInLog(app app, streamlet string) error {
+	logs, err := getLogs(swissKnifeApp, streamlet)
+	if err != nil {
+		return err
+	}
+	lastLine, err := getLastMinOneLine(logs)
+	if err != nil {
+		return err
+	}
+	Expect(lastLine).To(ContainSubstring("count:"))
+	return nil
+}
+
+func getToken() string {
+	data, err := ioutil.ReadFile(JsonTokenSrc)
+	if err != nil {
+		Fail("Can't read credentials for test")
+	}
+	return string(data)
+}
+
+func deploy(app app, pwd string) (deployRes string, deployErr error) {
+	cmd := exec.Command("kubectl", "cloudflow", "deploy", app.image, "--username", "_json_key", "--password", pwd)
+	out, err := cmd.CombinedOutput()
+	return string(out), err
+}
+
+func fakeDeploy(app app, pwd string) (deployRes string, deployErr error) {
+	fmt.Println(app.name)
+	fmt.Println(pwd[1:1])
+	return "OK", nil
+}
+
+type appListEntry struct {
+	name         string
+	namespace    string
+	version      string
+	creationtime string
+}
+
+type podEntry struct {
+	name     string
+	ready    string
+	status   string
+	restarts string
+	age      string
+}
+
+type streamletPod struct {
+	streamlet string
+	pod       string
+	status    string
+	restarts  int
+	ready     bool
+}
+
+type appStatus struct {
+	name          string
+	namespace     string
+	version       string
+	created       string
+	status        string
+	streamletPods []streamletPod
+}
+
+// func status(app app) (status appStatus, err error )
+
+func listAppNames() (entries []string, err error) {
+	apps, er := listApps()
+	if er != nil {
+		err = er
+		return
+	}
+	list := make([]string, len(apps))
+	for _, entry := range apps {
+		list = append(list, entry.name)
+	}
+	return list, nil
+}
+
+func listApps() (entries []appListEntry, err error) {
+	cmd := exec.Command("kubectl", "cloudflow", "list")
+	out, er := cmd.CombinedOutput()
+	if er != nil {
+		log.Fatal("could not check app status")
+		err = er
+		return
+	}
+	str := string(out)
+	splits := strings.Split(str, "\n")
+	whitespaces := regexp.MustCompile(`\s+`)
+	var res []appListEntry
+	for i, line := range splits {
+		switch i {
+		case 0, 1:
+			continue
+		default:
+			parts := whitespaces.Split(line, AllSubstrings)
+			if len(parts) == 7 {
+				appEntry := appListEntry{parts[0], parts[1], parts[2], parts[3] + parts[4] + parts[5] + parts[6]}
+
+				res = append(res, appEntry)
+			}
+		}
+	}
+	return res, nil
+}
+
+func ensureAppNotDeployed(app app) error {
+	apps, err := listApps()
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, entry := range apps {
+		if entry.name == app.name {
+			fmt.Printf("This is the app you are looking for: [%s]", entry.name)
+			found = true
+		}
+	}
+	if found {
+		err := undeploy(app)
+		if err != nil {
```

Comment:
You could make this a tiny amount less annoying by replacing these two lines with ``if err:= undeploy(app); err != nil {``